clang/lib/DPCT/ASTTraversal.cpp

```diff
@@ -4639,305 +4626,83 @@ void RandomFunctionCallRule::runRule(const MatchFinder::MatchResult &Result) {
-
-  bool CanAvoidUsingLambda = false;
-  SourceLocation OuterInsertLoc;
-  std::string OriginStmtType;
-  bool NeedUseLambda = isConditionOfFlowControl(
-      CE, OriginStmtType, CanAvoidUsingLambda, OuterInsertLoc);
-  bool IsInReturnStmt = isInReturnStmt(CE, OuterInsertLoc);
-  bool CanAvoidBrace = false;
-  const CompoundStmt *CS = findImmediateBlock(CE);
-  if (CS && (CS->size() == 1)) {
-    const Stmt *S = *(CS->child_begin());
-    if (CE == S || dyn_cast<ReturnStmt>(S))
-      CanAvoidBrace = true;
-  }
+  SourceLocation PrefixInsertLoc = SR.getBegin();
 
-  if (NeedUseLambda || IsMacroArg || IsInReturnStmt) {
-    NeedUseLambda = true;
-    SourceRange SR = getFunctionRange(CE);
-    PrefixInsertLoc = SR.getBegin();
-    SuffixInsertLoc = SR.getEnd();
-    if (IsInReturnStmt) {
-      CanAvoidUsingLambda = true;
-      OriginStmtType = "return";
-    }
-  }
-
-  std::string IndentStr = getIndent(PrefixInsertLoc, SM).str();
-  std::string PrefixInsertStr;
-
-  std::string Msg = "this call is redundant in SYCL.";
-  if (FuncName == "curandSetPseudoRandomGeneratorSeed" ||
-      FuncName == "curandSetQuasiRandomGeneratorDimensions") {
-    if (IsAssigned) {
-      report(PrefixInsertLoc, Diagnostics::FUNC_CALL_REMOVED_0, false, FuncName,
-             Msg);
-      emplaceTransformation(new ReplaceStmt(CE, false, "0"));
-    } else {
-      report(PrefixInsertLoc, Diagnostics::FUNC_CALL_REMOVED, false, FuncName,
-             Msg);
-      emplaceTransformation(new ReplaceStmt(CE, false, ""));
-    }
+  if (IsAssigned) {
+    report(PrefixInsertLoc, Diagnostics::NOERROR_RETURN_COMMA_OP, false);
+    insertAroundStmt(CE, "(", ", 0)");
   }
 
   if (FuncName == "curandCreateGenerator" ||
       FuncName == "curandCreateGeneratorHost") {
-    bool IsDuplicated = true;
-    auto REInfo = DpctGlobalInfo::getInstance().findRandomEngine(CE->getArg(0));
-    if (!REInfo) {
-      DpctGlobalInfo::getInstance().insertRandomEngine(CE->getArg(0));
-      REInfo = DpctGlobalInfo::getInstance().findRandomEngine(CE->getArg(0));
-      IsDuplicated = false;
-    }
-
     std::string EnumStr = ExprAnalysis::ref(CE->getArg(1));
     if (MapNames::RandomEngineTypeMap.find(EnumStr) ==
         MapNames::RandomEngineTypeMap.end()) {
-      report(SM.getExpansionLoc(REInfo->getDeclaratorDeclTypeBeginLoc()),
-             Diagnostics::UNMIGRATED_TYPE, false, "curandGenerator_t",
-             "the migration depends on the second argument of " + FuncName);
       report(PrefixInsertLoc, Diagnostics::NOT_SUPPORTED_PARAMETER, false,
              FuncName, "parameter " + EnumStr + " is unsupported");
       return;
     }
-    REInfo->setUnsupportEngineFlag(false);
-
-    if (EnumStr == "CURAND_RNG_PSEUDO_XORWOW" ||
-        EnumStr == "CURAND_RNG_QUASI_SOBOL64" ||
-        EnumStr == "CURAND_RNG_QUASI_SCRAMBLED_SOBOL64") {
-      report(SM.getExpansionLoc(REInfo->getDeclaratorDeclTypeBeginLoc()),
```

Comment:
Diagnostics::DIFFERENT_GENERATOR and Diagnostics::DIFFERENT_BASIC_GENERATOR should be emitted after this change.

Reply:
done